Tutorials/Python_Developer_Tutorials/foundation/Commanding-Local/Proxy-Local-Commands-Tutorial_.py

```diff
@@ -1,87 +1,110 @@
-time.sleep(1) #Delay so it is obvious that both LED's turn on at the same time in the next command
+time.sleep(1)  # Delay so it is obvious that both LED's turn on at the same time in the next command
 
-#Turn Both LED 1 and LED 2 ON simultaneously
-print "Turning ON both the Green and Red LED (LED #1 + LED #2)"
-command = faraday_cmd.CommandLocalGPIO((gpioallocations.LED_1|gpioallocations.LED_2), 0, 0, 0, 0, 0)
+# Turn Both LED 1 and LED 2 ON simultaneously
+print "Turning ON both the Green and Red LED (LED #1 | LED #2)"
+command = faraday_cmd.CommandLocalGPIO((gpioallocations.LED_1 | gpioallocations.LED_2), 0, 0, 0, 0, 0)
 faraday_1.POST(local_device_callsign, local_device_node_id, faraday_1.CMD_UART_PORT, command)
 time.sleep(1)
 
-#Turn Both LED 1 and LED 2 OFF simultaneously
+# Turn Both LED 1 and LED 2 OFF simultaneously
 print "Turning Off both the Green and Red LED (LED #1 + LED #2)"
-command = faraday_cmd.CommandLocalGPIO(0, 0, 0, (gpioallocations.LED_1|gpioallocations.LED_2), 0, 0)
+command = faraday_cmd.CommandLocalGPIO(0, 0, 0, (gpioallocations.LED_1 | gpioallocations.LED_2), 0, 0)
 faraday_1.POST(local_device_callsign, local_device_node_id, faraday_1.CMD_UART_PORT, command)
 
+
 ###############
 ## ECHO MESSAGE
 ###############
-print "/n** Beginning ECHO command test** /n"
-#Use the general command library to send a text message to the Faraday UART "ECHO" command. Will only ECHO a SINGLE packet. This will send the payload of the message back (up to 62 bytes, this can be updated in firmware to 124!)
-originalmsg = "This will ECHO back on UART" #Cannot be longer than max UART payload size!
+print "\n** Beginning ECHO command test** \n"
+originalmsg = "This will ECHO back on UART"  # Cannot be longer than max UART Transport layer payload size!
+#  Create command packet
 command = faradaycommands.commandmodule.create_command_datagram(faraday_cmd.CMD_ECHO, originalmsg)
+#  Send command
 faraday_1.POST(local_device_callsign, local_device_node_id, faraday_1.CMD_UART_PORT, command)
-
-#Retrive waiting data packet in UART Transport service number for the COMMAND application (Use GETWait() to block until ready or return False).
-rx_echo_raw = faraday_1.GETWait(local_device_callsign, local_device_node_id, faraday_1.CMD_UART_PORT, sec_timeout = 3)  # Wait for up to 3 seconds for data to arrive
-
-#Now parse data again
-b64_data = rx_echo_raw[0]['data']
-echo_decoded = faraday_1.DecodeRawPacket(b64_data)
-
-#Display information
-print "Original Message: ", originalmsg
-print "RAW Received BASE64 ECHO'd Message:", b64_data
-print "Decoded received ECHO'd Message:", echo_decoded #Note that ECHO sends back a fixed packed regardless. Should update to send back exact length.
-
-
+#  Retrive waiting data packet in UART Transport service number for the COMMAND application (Use GETWait() to block until ready or timeout
+rx_echo_raw = faraday_1.GETWait(local_device_callsign, local_device_node_id, faraday_1.CMD_UART_PORT,
```

Comment:
This should be checked for the "error" JSON response as shown below:
![image](https://cloud.githubusercontent.com/assets/331540/23398334/fd3c5fe4-fd50-11e6-85fe-8a1aee975197.png)

failing gracefully and printing the error message returned by proxy is what we should do. This is necessary in all applications but just noticed that we weren't using this functionality I built in to the API.

This is failing on first try because the GETWait() because you are trying to obtain data from the CMD_UART_PORT for the first time. All prior LED operations were a POST() to the port. This means that the buffer hasn't been created yet for this particular port on the RX side of things. This is Issue #60 creeping up.